### ISSUE TITLE
DSL.inline(array) renders bind value markers in BigQuery

### ISSUE THREAD

**Issue by zineddine6** [jOOQ/jOOQ#17079]
### Expected behavior

When I create an array with jOOQ I expect the SQL generated to be: `array[ my_value ]` 

### Actual behavior

Instead, I get this: `array [ cast(? as string) ]`

### Steps to reproduce the problem

```
String[] array = {"a"};
Field<String[]> jooq_array = inline(array);
```
this will generate:
```
array [ cast(? as string) ]
```
which is not supported by BQ as it throws the following error:
`com.google.cloud.bigquery.BigQueryException: Positional parameters are not supported`

### jOOQ Version

Entreprise edition 3.20-SNAPSHOT

### Database product and version

Latest BigQuery

### Java Version

openJDK 17.0.8.1 2023-08-24

**Comment by lukaseder** [comment]
Thanks a lot for the report. This is a new bug, introduced with the recently added BigQuery array support:

- https://github.com/jOOQ/jOOQ/issues/17037

I can see how this happens. The `inline()` flag isn't propagated from the array value to its elements. I'll fix this right away.

**Comment by lukaseder** [comment]
A workaround is to use static statements:
https://www.jooq.org/doc/latest/manual/sql-execution/statement-type/

**Comment by lukaseder** [comment]
I see now. Our integration tests regarding array bind values only checked using the statement type, not using explicit inline values.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.20.0. Thanks again for your report!

**Comment by lukaseder** [comment]
I've deployed a new 3.20.0-SNAPSHOT version including this fix
